addons/hr_holidays/models/hr_employee.py

```diff
@@ -76,16 +76,17 @@ def _compute_remaining_leaves(self):
             employee.remaining_leaves = value
 
     def _compute_allocation_count(self):
+        data = self.env['hr.leave.allocation'].read_group([
+            ('employee_id', 'in', self.ids),
+            ('holiday_status_id.active', '=', True),
+            ('state', '=', 'validate'),
+            '|',
+                ('date_to', '=', False),
+                ('date_to', '>=', datetime.date.today()),
+        ], ['number_of_days:sum', 'employee_id'], ['employee_id'])
+        rg_results = dict((d['employee_id'][0], d['number_of_days']) for d in data)
         for employee in self:
-            allocations = self.env['hr.leave.allocation'].search([
-                ('employee_id', '=', employee.id),
-                ('holiday_status_id.active', '=', True),
-                ('state', '=', 'validate'),
-                '|',
-                    ('date_to', '=', False),
-                    ('date_to', '>=', datetime.date.today()),
-            ])
-            employee.allocation_count = sum(allocations.mapped('number_of_days'))
+            employee.allocation_count = rg_results.get(employee.id, 0.0)
```

Comment:
NB: `allocation_count` doesn't seem used, the field could probably be removed, to keep only the `allocation_display` field